src/sentry/static/sentry/app/components/sidebar/index.jsx

```diff
@@ -86,11 +76,9 @@ class Sidebar extends React.Component {
   }
 
   // Sidebar doesn't use children, so don't use it to compare
-  // Also ignore location, will re-render when routes change (instead of query params)
-  shouldComponentUpdate({children, location, ...nextPropsToCompare}, nextState) {
+  shouldComponentUpdate({children, ...nextPropsToCompare}, nextState) {
     const {
       children: _children, // eslint-disable-line no-unused-vars
-      location: _location, // eslint-disable-line no-unused-vars
```

Comment:
Can we preserve this behavior and always link to the index page, and handle navigation in the on click handler instead? 

This will:
* still result in an additional router state change if the user opens the link in a new tab (or copies the link from context menu)
  * I think that's fine as a trade-off since sidebar re-renders happen much more frequently
* display the index URL without URL params in browser status bar on hover
  * it's not really human parseable anyway

Reply:
I'm not sure exactly what you mean by this or how it would work since onClick functions will be run before the transition? I think I'd rather keep all the logic responsible for determining what the new URL will be in a single place rather than split across onClick and href.

Reply:
Yeah I was just thinking that onClick hijacks the link and we can `router.push` the url with query string and just leave the urls alone for the `SidebarItems`. This way we can keep the sidebar from re-rendering.